src/cdsetool/download.py

```diff
@@ -126,3 +143,60 @@ def _set_proxy(options, session):
     if proxies != {}:
         session.proxies.update(proxies)
     return session
+
+
+def validity_check(temp_path, product_info):
+    """Given it's temp_path and metadata info checks if the data downloaded is valid"""
+    size = os.path.getsize(temp_path)
+    content_length = (
+        product_info.get("properties").get("services").get("download").get("size")
+    )
+    if size > content_length:
+        # Dowloaded more than the size of the file.
+        return Validity.INVALID
+    if size == content_length:
+        checksum_comparison = _checksum_compare(temp_path, product_info)
+        if checksum_comparison is None:
+            # No data available for the checksum
+            return Validity.IGNORE
+        if checksum_comparison:
+            return Validity.VALID
+        # Checksum failed
+        return Validity.INVALID
+    # here it's a partial download logic to implement is TODO
+    return Validity.CONTINUE
+
+
+def _checksum_compare(temp_path, product_info):
+    """Compare a given MD5 checksum with one calculated from a file."""
+    checksum = None
+    algo = None
+    checksum_list = product_info.get("Checksum", [])
+    algo, checksum = from_checksum_list_to_checksum(checksum_list)
+    if checksum is None:
+        # no checksum available
+        return None
+    with open(temp_path, "rb") as f:
+        while True:
+            block_data = f.read(8192)
+            if not block_data:
+                break
+            algo.update(block_data)
+    return algo.hexdigest().lower() == checksum.lower()
```

Comment:
Should the `.lower()` really be there?

I think changing line 178 to `return Validity.IGNORE` and this line to `return Validity.VALID if algo.hexdigest().lower() == checksum.lower() else Validity.INVALID` would eliminate the need for the logic on lines 159-165.